### ISSUE TITLE
Conversations as Distributor for UnifiedPush via XMPP

### ISSUE THREAD

**Issue by mcg-android** [iNPUTmice/Conversations#4216]
Using Conversations is clearly the most energy-efficient instant messaging solution currently available for Android users trying to avoid Google/Firebase Cloud Messaging.

Do you think it would be feasible to extend the Conversations app to be a UnifiedPush Distributor, talking to a UnifiedPush Provider via XMPP?

"UnifiedPush via XMPP" would make for a reliable, low-energy approach. The UnifiedPush Provider could get implemented as an extension for XMPP server software or even as an XMPP client itself.

**Comment by iNPUTmice** [comment]
This is interesting from a technical standpoint and there will probably be some experiments once Ltt.rs is out and Conversations 3.x is done

**Comment by mcg-android** [comment]
Thanks!

I'll keep an eye on https://ltt.rs and https://github.com/iNPUTmice/Conversations/releases, and I guess I'll later reopen this issue, if I may.

**Comment by xundeenergie** [comment]
unifiedpush would be really great.

you "just" need to integrate the libraries in the app, and then a user can choose gotify-uo or nextpush as notification-provider and setup the notification-server (a gotify-server + gateway-proxy or a nextcloud-app like https://github.com/UP-NextPush/server-app).

Please think abour!!

**Comment by iNPUTmice** [comment]
The next version of Conversations (2.12.0) will act as a UnifiedPush Distributor.

**Comment by licaon-kter** [comment]
A fun read https://f-droid.org/2022/12/18/unifiedpush.html

**Comment by mcg-android** [comment]
> The next version of Conversations (2.12.0) will act as a UnifiedPush Distributor.

Wait, what, it's there, just like that? I'm excited!

So I upgraded the Conversations app to version 2.12.1+free, went into its settings, selected the presumably most reliable XMPP account for the "UnifiedPush Distributor" functionality, and left "Push Server" set to "up.conversations.im" (which, too, presumably is very reliable :-)).

I then changed the settings of my most energy-hungry instant-messaging app to no longer use its own polling mechanism but instead use UnifiedPush via Conversations.

Here's the result in terms of battery usage — see where the graph went almost flat a few hours ago? That's Conversations saving battery, it's phantastic!
![battery-UP_50p](https://user-images.githubusercontent.com/70754523/212888459-ebf84854-5f15-4153-9298-e2c788cc8cb8.png)
